migration/1664367797442-changeMainCategoriesAndSubCategories.ts

```diff
@@ -0,0 +1,223 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+const updateSubCategory = async (
+  queryRunner: QueryRunner,
+  params: {
+    mainCategorySlug: string;
+    newName: string;
+    oldName: string;
+    newValue: string;
+  },
+) => {
+  const mainCategory = (
+    await queryRunner.query(`SELECT * FROM main_category
+        WHERE slug='${params.mainCategorySlug}'`)
+  )[0];
+
+  await queryRunner.query(`
+           UPDATE category SET name='${params.newName}', value='${params.newValue}', "mainCategoryId"=${mainCategory.id} WHERE name='${params.oldName}'
+        `);
+};
+
+export class changeMainCategoriesAndSubCategories1664367797442
+  implements MigrationInterface
+{
+  // based on https://docs.google.com/spreadsheets/d/1DAB5UHCYO4hkw_RLRtz8DLXpGduVDRMHfiW-71H0m8o/edit#gid=122470553
+  // and https://github.com/Giveth/GIVeconomy/issues/723
+  async up(queryRunner: QueryRunner): Promise<void> {
+    const lastMainCategory = (
+      await queryRunner.query(
+        `
+              SELECT * FROM main_category
+              ORDER BY id DESC
+            `,
+      )
+    )[0];
+    let lastId = lastMainCategory.id;
+    await queryRunner.query(`
+           INSERT INTO public.main_category(id, title, slug, description, banner) VALUES
+             (${++lastId}, 'Nature','nature', '', ''),
+             (${++lastId}, 'Community','community', '', ''),
+             (${++lastId}, 'Finance','finance', '', ''),
+             (${++lastId}, 'Education','education', '', ''),
+             (${++lastId}, 'Equality','equality', '', ''),
+             (${++lastId}, 'Other','other', '', '');
```

Comment:
Yeah at the beginning I didnt' pass the id, but I was getting this error, you can test it yourself in your system, I didn't know what was the reason but after passing the id it worked

<img width="1101" alt="Screen Shot 1401-07-10 at 12 10 58" src="https://user-images.githubusercontent.com/9850545/193445740-f408b54d-7faa-48ce-bdf4-1e352441dab2.png">

https://github.com/Giveth/impact-graph/actions/runs/3151490462/jobs/5125501150

Reply:
Let me do a quick test and get back to you. Thanks for mentioning this.

Reply:
So the reason it doesn't work is because we are defining the columns in the migrations with
```SQL
id SERIAL NOT NULL
```
There is no generationStrategy = 'increment'.
This is fixed later on when the server runs because we have typeorm synchronization.

Reply:
I tried to find a way to update the metadata of the column in the migration. But there is no easy fix. Let's leave it like this for now with the lastId.

For future migrations when creating tables we should add
```
isGenerated: true,
generationStrategy: 'increment',
```
By pure sql or with typeorm helpers.